lib/spack/docs/advanced_topics.rst

```diff
@@ -0,0 +1,263 @@
+.. Copyright Spack Project Developers. See COPYRIGHT file for details.
+
+   SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+.. _toolchains:
+
+=============================
+Defining and Using Toolchains
+=============================
+
+Spack can be configured to associate certain combinations of specs for
+easy reference on the command line and in config and environment
+files. These combinations are called ``toolchains``, because their
+primary intended use is for associating compiler combinations to
+apply. Toolchains are referenced by name like a direct dependency,
+using the ``%`` sigil. There are two styles of toolchain config, one
+using conditional dependencies through the spec syntax and one with
+conditionals explicitly in the yaml:
+
+.. code-block:: yaml
+
+   toolchains:
+     gcc_all: cflags=-O3 '%[when=%c virtuals=c]gcc %[when=%cxx virtuals=cxx]gcc %[when=%fortran virtuals=fortran]gcc'
+     llvm_gfortran:
+     - spec: cflags=-O3
+     - spec: '%[virtuals=c]llvm'
+       when: '%c'
+     - spec: '%[virtuals=cxx]llvm'
+       when: '%cxx'
+     - spec: '%[virtuals=fortran]gcc'
+       when: '%fortran'
```

Comment:
why document both styles?  I would start with the simple style and the `%c=llvm` syntax, and then document the longer syntax later (if at all)... I don't think it's interesting to users that there are two ways to do this.